src/lib.rs

```diff
@@ -49,52 +42,8 @@ struct ScrollableCache {
     split_points: Vec<(usize, Pos2, Pos2)>,
 }
 
-#[derive(Default)]
-struct ImageHandleCache {
-    cache: HashMap<String, Promise<Result<TextureHandle, String>>>,
-}
-
-impl ImageHandleCache {
-    fn clear(&mut self) {
-        self.cache.clear();
-    }
-
-    fn load(&mut self, ctx: &egui::Context, url: String) -> Poll<Result<TextureHandle, String>> {
-        let promise = self.cache.entry(url.clone()).or_insert_with(|| {
-            let ctx = ctx.clone();
-            let (sender, promise) = Promise::new();
-            fetch_data::get_image_data(&url.clone(), move |result| {
-                match result {
-                    Ok(bytes) => {
-                        sender.send(parse_image(&ctx, &url, &bytes));
-                    }
-
-                    Err(err) => {
-                        sender.send(Err(err));
-                    }
-                };
-                ctx.request_repaint();
-            });
-            promise
-        });
-
-        promise.poll().map(|r| r.clone())
-    }
-
-    fn loaded(&self) -> impl Iterator<Item = &TextureHandle> {
-        self.cache
-            .values()
-            .flat_map(|p| p.ready())
-            .flat_map(|r| r.as_ref().ok())
-    }
-}
-
 /// A cache used for storing content such as images.
 pub struct CommonMarkCache {
-    // Everything stored here must take into account that the cache is for multiple
-    // CommonMarkviewers with different source_ids.
```

Comment:
The comment is for the entire struct and not for the image field. It's just badly placed by me.

Reply:
I added it back in